ddsketch/build.rs

```diff
@@ -0,0 +1,59 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present
+// Datadog, Inc.
+
+use std::error::Error;
+
+#[cfg(feature = "generate-protobuf")]
+use std::{
+    env,
+    fs::File,
+    io::{Read, Write},
+    path::Path,
+};
+
+#[cfg(feature = "generate-protobuf")]
+const HEADER: &str =
+    "// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present
+// Datadog, Inc.
+
+// This file has been automatically generated from build.rs
+
+";
+
+fn main() -> Result<(), Box<dyn Error>> {
+    #[cfg(feature = "generate-protobuf")]
+    {
+        // protoc is required to compile proto files. This uses protobuf_src to compile protoc
```

Comment:
is this automatically handle by cargo? Do we need to improve internal or public (e.g in repo) doc to ease building locally?

Reply:
If the proto files were to be modified, you would need to `cargo build --feature generate-protobuf` to generate the corresponding rust code, but it adds significant compile time, so it's hidden behind a feature.

It's the same mechanism as the tracer-payloads crate in this repo btw